shopfloor/models/stock_location.py

```diff
@@ -35,3 +36,26 @@ def _get_reserved_move_lines(self):
     def _compute_reserved_move_lines(self):
         for rec in self:
             rec.update({"reserved_move_line_ids": rec._get_reserved_move_lines()})
+
+    def planned_qty_in_location_is_empty(self):
+        """Return if a location will be empty when move lines will be confirmed
+
+        Used for the "zero check". We need to know if a location is empty, but since
+        we set the move lines to "done" only at the end of the unload workflow, we
+        have to look at the qty_done of the move lines from this location.
+        """
+        self.ensure_one()
+        quants = self.env["stock.quant"].search(
+            [("quantity", ">", 0), ("location_id", "=", self.id)]
+        )
+        remaining = sum(quants.mapped("quantity"))
+        lines = self.env["stock.move.line"].search(
+            [
+                ("state", "!=", "done"),
```

Comment:
Probably, but as `state` is a related on `move_id.state`, and that we get no move lines on a draft/cancel move, it's maybe useless. But I agree it would be clearer by doing so.